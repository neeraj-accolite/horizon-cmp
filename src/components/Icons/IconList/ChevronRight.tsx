import React from 'react';
import { IconProps } from '../Icons.model';

const ChevronRight: React.FC<IconProps> = ({
  fill = 'currentColor',
  filled,
  size,
  height,
  width,
  ...props
}) => {
  return (
    <svg
      data-testid="chevron-right-icon"
      fill={filled ? fill : 'none'}
      height={height || size || 25}
      viewBox="0 0 24 25"
      width={width || size || 24}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <mask
        id="mask0"
        style={{ maskType: 'luminance' }}
        maskUnits="userSpaceOnUse"
        x="8"
        y="6"
        width="8"
        height="13"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M9.99984 6.7L8.58984 8.11L13.1698 12.7L8.58984 17.29L9.99984 18.7L15.9998 12.7L9.99984 6.7Z"
          fill="white"
        />
      </mask>
      <g mask="url(#mask0)">
        <rect x="-13" y="-12.3" width="50" height="50" fill={fill} />
        <mask
          id="mask1"
          style={{ maskType: 'luminance' }}
          maskUnits="userSpaceOnUse"
          x="-13"
          y="-13"
          width="50"
          height="51"
        >
          <rect x="-13" y="-12.3" width="50" height="50" fill="white" />
        </mask>
        <g mask="url(#mask1)"></g>
      </g>
    </svg>
  );
};

export default ChevronRight;
