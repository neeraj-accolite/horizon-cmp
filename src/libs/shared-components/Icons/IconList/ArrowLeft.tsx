import React from 'react';
import { IconProps } from '../Icons.model';

const ArrowLeft: React.FC<IconProps> = ({
  fill = 'currentColor',
  filled,
  size,
  height,
  width,
  ...props
}) => {
  return (
    <svg
      data-testid="arrow-left-icon"
      fill="currentColor"
      height={height || size || 24}
      viewBox="0 0 24 24"
      width={width || size || 24}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M14.9998 19.9201L8.47984 13.4001C7.70984 12.6301 7.70984 11.3701 8.47984 10.6001L14.9998 4.08008"
        stroke={fill}
        strokeWidth="1.5"
        strokeMiterlimit="10"
        strokeLinecap="round"
        strokeLinejoin="round"
        fill={filled ? fill : 'none'}
      />
    </svg>
  );
};

export default ArrowLeft;
