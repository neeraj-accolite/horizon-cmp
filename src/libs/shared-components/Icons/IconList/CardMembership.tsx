import React from 'react';
import { IconProps } from '../Icons.model';

const CardMembership: React.FC<IconProps> = ({
  fill = 'currentColor',
  size,
  height,
  width,
  ...props
}) => {
  return (
    <svg
      data-testid="card-membership-icon"
      fill="none"
      height={height || size}
      viewBox="0 0 33 32"
      width={width || size}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <mask
        id="mask0_5004_7798"
        style={{ maskType: 'luminance' }}
        maskUnits="userSpaceOnUse"
        x="3"
        y="2"
        width="27"
        height="28"
      >
        <path
          fillRule="evenodd"
          clipRule="evenodd"
          d="M27 2.66663H5.66667C4.18667 2.66663 3 3.85329 3 5.33329V20C3 21.48 4.18667 22.6666 5.66667 22.6666H11V29.3333L16.3333 26.6666L21.6667 29.3333V22.6666H27C28.48 22.6666 29.6667 21.48 29.6667 20V5.33329C29.6667 3.85329 28.48 2.66663 27 2.66663ZM27 20H5.66667V17.3333H27V20ZM27 13.3333H5.66667V5.33329H27V13.3333Z"
          fill="white"
        />
      </mask>
      <g mask="url(#mask0_5004_7798)">
        <rect
          x="-17"
          y="-17.3334"
          width="66.6667"
          height="66.6667"
          fill={fill}
        />
        <mask
          id="mask1"
          style={{ maskType: 'luminance' }}
          maskUnits="userSpaceOnUse"
          x="-17"
          y="-18"
          width="67"
          height="68"
        >
          <rect
            x="-17"
            y="-17.3334"
            width="66.6667"
            height="66.6667"
            fill="white"
          />
        </mask>
        <g mask="url(#mask1)"></g>
      </g>
    </svg>
  );
};

export default CardMembership;
