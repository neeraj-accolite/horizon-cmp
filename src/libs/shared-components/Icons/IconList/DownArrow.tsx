import React from 'react';
import { IconProps } from '../Icons.model';

const DownArrow: React.FC<IconProps> = ({
  fill = 'currentColor',
  filled,
  size,
  ...props
}) => {
  return (
    <svg
      data-testid="down-arrow-icon"
      fill="currentColor"
      height={size || 24}
      viewBox="0 0 24 24"
      width={size || 24}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M19.9201 8.9502L13.4001 15.4702C12.6301 16.2402 11.3701 16.2402 10.6001 15.4702L4.08008 8.9502"
        stroke={fill}
        strokeWidth="1.5"
        strokeMiterlimit="10"
        strokeLinecap="round"
        strokeLinejoin="round"
        fill={filled ? fill : 'none'}
      />
    </svg>
  );
};

export default DownArrow;
