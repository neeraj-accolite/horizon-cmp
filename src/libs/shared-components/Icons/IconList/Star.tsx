import React from 'react';
import { IconProps } from '../Icons.model';

const Star: React.FC<IconProps> = ({
  fill = 'currentColor',
  filled,
  size,
  height,
  width,
  ...props
}) => {
  return (
    <svg
      data-testid="star-icon"
      fill={filled ? fill : 'none'}
      height={height || size || 16}
      viewBox="0 0 16 16"
      width={width || size || 16}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M11.3038 13.4049C11.5163 13.2442 11.563 12.9799 11.4334 12.6067L10.3502 9.37774L13.1179 7.39268C13.4392 7.16463 13.5688 6.92104 13.4859 6.66707C13.4029 6.41311 13.1593 6.28872 12.7603 6.2939L9.36544 6.31982L8.33403 3.0753C8.20964 2.69695 8.02306 2.5 7.75355 2.5C7.48922 2.5 7.30263 2.69695 7.17824 3.0753L6.14684 6.31982L2.75202 6.2939C2.35294 6.28872 2.10934 6.41311 2.02641 6.66707C1.9383 6.92104 2.07306 7.16463 2.3944 7.39268L5.16208 9.37774L4.07885 12.6067C3.94928 12.9799 3.99592 13.2442 4.20842 13.4049C4.42092 13.5707 4.69044 13.5137 5.01178 13.2805L7.75355 11.2643L10.5005 13.2805C10.8218 13.5137 11.0862 13.5707 11.3038 13.4049Z"
        fill={fill}
      />
    </svg>
  );
};

export default Star;
