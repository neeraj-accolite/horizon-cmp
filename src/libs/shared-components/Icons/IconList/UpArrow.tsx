import React from 'react';
import { IconProps } from '../Icons.model';

const UpArrow: React.FC<IconProps> = ({
  fill = 'currentColor',
  filled,
  size,
  ...props
}) => {
  return (
    <svg
      data-testid="up-arrow-icon"
      fill="none"
      height={size || 25}
      viewBox="0 0 25 25"
      width={size || 25}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M20.4201 15.8197L13.9001 9.29967C13.1301 8.52967 11.8701 8.52967 11.1001 9.29967L4.58008 15.8197"
        stroke={fill}
        strokeWidth="1.5"
        strokeMiterlimit="10"
        strokeLinecap="round"
        strokeLinejoin="round"
        fill={filled ? fill : 'none'}
      />
    </svg>
  );
};

export default UpArrow;
