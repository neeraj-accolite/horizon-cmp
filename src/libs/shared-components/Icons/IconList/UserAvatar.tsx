import React from 'react';
import { IconProps } from '../Icons.model';

const UserAvatar: React.FC<IconProps> = ({
  fill = 'currentColor',
  filled = 'true',
  size,
  ...props
}) => {
  return (
    <svg
      data-testid="user-avatar-icon"
      fill="currentColor"
      height={size || 24}
      viewBox="0 0 24 24"
      width={size || 24}
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        fillRule="evenodd"
        clipRule="evenodd"
        d="M12 22.5C17.7441 22.5 22.5 17.7441 22.5 12C22.5 6.26618 17.7338 1.5 11.9897 1.5C6.25588 1.5 1.5 6.26618 1.5 12C1.5 17.7441 6.26618 22.5 12 22.5ZM18.5265 17.8574C17.4868 16.6221 15.0985 15.5103 12 15.5103C8.90147 15.5103 6.52353 16.6118 5.47353 17.8471C4.09412 16.3029 3.26029 14.2544 3.26029 12C3.26029 7.15147 7.13088 3.25 11.9897 3.25C16.8485 3.25 20.7397 7.15147 20.75 12C20.7603 14.2647 19.9162 16.3132 18.5265 17.8574ZM12 13.7603C13.9765 13.7809 15.5309 12.0926 15.5309 9.87941C15.5309 7.8 13.9765 6.07059 12 6.07059C10.0235 6.07059 8.45882 7.8 8.46912 9.87941C8.47941 12.0926 10.0235 13.7397 12 13.7603Z"
        fill={filled ? fill : 'none'}
        stroke={fill}
      />
    </svg>
  );
};

export default UserAvatar;
