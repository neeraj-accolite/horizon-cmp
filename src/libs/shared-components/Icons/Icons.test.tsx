import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import ChevronRight from './IconList/ChevronRight';
import LocalOffer from './IconList/LocalOffer';
import UserAvatar from './IconList/UserAvatar';
import DownArrow from './IconList/DownArrow';
import ArrowLeft from './IconList/ArrowLeft';
import Star from './IconList/Star';
import CardMembership from './IconList/CardMembership';
import UpArrow from './IconList/UpArrow';

describe('Icons', () => {
  it('renders ChevronRight icon with filled prop true', () => {
    const { container } = render(
      <ChevronRight fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders ChevronRight icon with filled prop false', () => {
    const { container } = render(
      <ChevronRight fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders LocalOffer icon with filled prop true', () => {
    const { container } = render(
      <LocalOffer fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders LocalOffer icon with filled prop false', () => {
    const { container } = render(
      <LocalOffer fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders UserAvatar icon with filled prop true', () => {
    const { container } = render(
      <UserAvatar fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders UserAvatar icon with filled prop false', () => {
    const { container } = render(
      <UserAvatar fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders DownArrow icon with filled prop true', () => {
    const { container } = render(
      <DownArrow fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders DownArrow icon with filled prop false', () => {
    const { container } = render(
      <DownArrow fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders ArrowLeft icon with filled prop true', () => {
    const { container } = render(
      <ArrowLeft fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders ArrowLeft icon with filled prop false', () => {
    const { container } = render(
      <ArrowLeft fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders Star icon with filled prop true', () => {
    const { container } = render(
      <Star fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders Star icon with filled prop false', () => {
    const { container } = render(
      <Star fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders CardMembership icon with filled prop true', () => {
    const { container } = render(
      <CardMembership fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders CardMembership icon with filled prop false', () => {
    const { container } = render(
      <CardMembership fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders UpArrow icon with filled prop true', () => {
    const { container } = render(
      <UpArrow fill="currentColor" filled={true} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders UpArrow icon with filled prop false', () => {
    const { container } = render(
      <UpArrow fill="currentColor" filled={false} size={24} />,
    );
    expect(container).toMatchSnapshot();
  });
});
