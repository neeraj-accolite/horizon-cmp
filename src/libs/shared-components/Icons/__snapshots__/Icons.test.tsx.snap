// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icons renders ArrowLeft icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="arrow-left-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.9998 19.9201L8.47984 13.4001C7.70984 12.6301 7.70984 11.3701 8.47984 10.6001L14.9998 4.08008"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-miterlimit="10"
      stroke-width="1.5"
    />
  </svg>
</div>
`;

exports[`Icons renders ArrowLeft icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="arrow-left-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14.9998 19.9201L8.47984 13.4001C7.70984 12.6301 7.70984 11.3701 8.47984 10.6001L14.9998 4.08008"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-miterlimit="10"
      stroke-width="1.5"
    />
  </svg>
</div>
`;

exports[`Icons renders CardMembership icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="card-membership-icon"
    fill="none"
    height="24"
    viewBox="0 0 33 32"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="28"
      id="mask0_5004_7798"
      maskUnits="userSpaceOnUse"
      style="mask-type: luminance;"
      width="27"
      x="3"
      y="2"
    >
      <path
        clip-rule="evenodd"
        d="M27 2.66663H5.66667C4.18667 2.66663 3 3.85329 3 5.33329V20C3 21.48 4.18667 22.6666 5.66667 22.6666H11V29.3333L16.3333 26.6666L21.6667 29.3333V22.6666H27C28.48 22.6666 29.6667 21.48 29.6667 20V5.33329C29.6667 3.85329 28.48 2.66663 27 2.66663ZM27 20H5.66667V17.3333H27V20ZM27 13.3333H5.66667V5.33329H27V13.3333Z"
        fill="white"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask0_5004_7798)"
    >
      <rect
        fill="currentColor"
        height="66.6667"
        width="66.6667"
        x="-17"
        y="-17.3334"
      />
      <mask
        height="68"
        id="mask1"
        maskUnits="userSpaceOnUse"
        style="mask-type: luminance;"
        width="67"
        x="-17"
        y="-18"
      >
        <rect
          fill="white"
          height="66.6667"
          width="66.6667"
          x="-17"
          y="-17.3334"
        />
      </mask>
      <g
        mask="url(#mask1)"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons renders CardMembership icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="card-membership-icon"
    fill="none"
    height="24"
    viewBox="0 0 33 32"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="28"
      id="mask0_5004_7798"
      maskUnits="userSpaceOnUse"
      style="mask-type: luminance;"
      width="27"
      x="3"
      y="2"
    >
      <path
        clip-rule="evenodd"
        d="M27 2.66663H5.66667C4.18667 2.66663 3 3.85329 3 5.33329V20C3 21.48 4.18667 22.6666 5.66667 22.6666H11V29.3333L16.3333 26.6666L21.6667 29.3333V22.6666H27C28.48 22.6666 29.6667 21.48 29.6667 20V5.33329C29.6667 3.85329 28.48 2.66663 27 2.66663ZM27 20H5.66667V17.3333H27V20ZM27 13.3333H5.66667V5.33329H27V13.3333Z"
        fill="white"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask0_5004_7798)"
    >
      <rect
        fill="currentColor"
        height="66.6667"
        width="66.6667"
        x="-17"
        y="-17.3334"
      />
      <mask
        height="68"
        id="mask1"
        maskUnits="userSpaceOnUse"
        style="mask-type: luminance;"
        width="67"
        x="-17"
        y="-18"
      >
        <rect
          fill="white"
          height="66.6667"
          width="66.6667"
          x="-17"
          y="-17.3334"
        />
      </mask>
      <g
        mask="url(#mask1)"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons renders ChevronRight icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="chevron-right-icon"
    fill="none"
    height="24"
    viewBox="0 0 24 25"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="13"
      id="mask0"
      maskUnits="userSpaceOnUse"
      style="mask-type: luminance;"
      width="8"
      x="8"
      y="6"
    >
      <path
        clip-rule="evenodd"
        d="M9.99984 6.7L8.58984 8.11L13.1698 12.7L8.58984 17.29L9.99984 18.7L15.9998 12.7L9.99984 6.7Z"
        fill="white"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask0)"
    >
      <rect
        fill="currentColor"
        height="50"
        width="50"
        x="-13"
        y="-12.3"
      />
      <mask
        height="51"
        id="mask1"
        maskUnits="userSpaceOnUse"
        style="mask-type: luminance;"
        width="50"
        x="-13"
        y="-13"
      >
        <rect
          fill="white"
          height="50"
          width="50"
          x="-13"
          y="-12.3"
        />
      </mask>
      <g
        mask="url(#mask1)"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons renders ChevronRight icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="chevron-right-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 25"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="13"
      id="mask0"
      maskUnits="userSpaceOnUse"
      style="mask-type: luminance;"
      width="8"
      x="8"
      y="6"
    >
      <path
        clip-rule="evenodd"
        d="M9.99984 6.7L8.58984 8.11L13.1698 12.7L8.58984 17.29L9.99984 18.7L15.9998 12.7L9.99984 6.7Z"
        fill="white"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask0)"
    >
      <rect
        fill="currentColor"
        height="50"
        width="50"
        x="-13"
        y="-12.3"
      />
      <mask
        height="51"
        id="mask1"
        maskUnits="userSpaceOnUse"
        style="mask-type: luminance;"
        width="50"
        x="-13"
        y="-13"
      >
        <rect
          fill="white"
          height="50"
          width="50"
          x="-13"
          y="-12.3"
        />
      </mask>
      <g
        mask="url(#mask1)"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons renders DownArrow icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="down-arrow-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19.9201 8.9502L13.4001 15.4702C12.6301 16.2402 11.3701 16.2402 10.6001 15.4702L4.08008 8.9502"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-miterlimit="10"
      stroke-width="1.5"
    />
  </svg>
</div>
`;

exports[`Icons renders DownArrow icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="down-arrow-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19.9201 8.9502L13.4001 15.4702C12.6301 16.2402 11.3701 16.2402 10.6001 15.4702L4.08008 8.9502"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-miterlimit="10"
      stroke-width="1.5"
    />
  </svg>
</div>
`;

exports[`Icons renders LocalOffer icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="local-offer-icon"
    fill="none"
    height="24"
    viewBox="0 0 32 32"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="28"
      id="mask1_5004_7786"
      maskUnits="userSpaceOnUse"
      style="mask-type: luminance;"
      width="28"
      x="2"
      y="2"
    >
      <path
        clip-rule="evenodd"
        d="M28.5465 15.44L16.5465 3.43996C16.0665 2.95996 15.3998 2.66663 14.6665 2.66663H5.33317C3.8665 2.66663 2.6665 3.86663 2.6665 5.33329V14.6666C2.6665 15.4 2.95984 16.0666 3.45317 16.56L15.4532 28.56C15.9332 29.04 16.5998 29.3333 17.3332 29.3333C18.0665 29.3333 18.7332 29.04 19.2132 28.5466L28.5465 19.2133C29.0398 18.7333 29.3332 18.0666 29.3332 17.3333C29.3332 16.6 29.0265 15.92 28.5465 15.44ZM7.33317 9.33329C6.2265 9.33329 5.33317 8.43996 5.33317 7.33329C5.33317 6.22663 6.2265 5.33329 7.33317 5.33329C8.43984 5.33329 9.33317 6.22663 9.33317 7.33329C9.33317 8.43996 8.43984 9.33329 7.33317 9.33329Z"
        fill="white"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask1_5004_7786)"
    >
      <rect
        fill="currentColor"
        height="66.6667"
        width="66.6667"
        x="-17.3335"
        y="-17.3334"
      />
      <mask
        height="68"
        id="mask1"
        maskUnits="userSpaceOnUse"
        style="mask-type: luminance;"
        width="68"
        x="-18"
        y="-18"
      >
        <rect
          fill="white"
          height="66.6667"
          width="66.6667"
          x="-17.3335"
          y="-17.3334"
        />
      </mask>
      <g
        mask="url(#mask1)"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons renders LocalOffer icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="local-offer-icon"
    fill="none"
    height="24"
    viewBox="0 0 32 32"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <mask
      height="28"
      id="mask1_5004_7786"
      maskUnits="userSpaceOnUse"
      style="mask-type: luminance;"
      width="28"
      x="2"
      y="2"
    >
      <path
        clip-rule="evenodd"
        d="M28.5465 15.44L16.5465 3.43996C16.0665 2.95996 15.3998 2.66663 14.6665 2.66663H5.33317C3.8665 2.66663 2.6665 3.86663 2.6665 5.33329V14.6666C2.6665 15.4 2.95984 16.0666 3.45317 16.56L15.4532 28.56C15.9332 29.04 16.5998 29.3333 17.3332 29.3333C18.0665 29.3333 18.7332 29.04 19.2132 28.5466L28.5465 19.2133C29.0398 18.7333 29.3332 18.0666 29.3332 17.3333C29.3332 16.6 29.0265 15.92 28.5465 15.44ZM7.33317 9.33329C6.2265 9.33329 5.33317 8.43996 5.33317 7.33329C5.33317 6.22663 6.2265 5.33329 7.33317 5.33329C8.43984 5.33329 9.33317 6.22663 9.33317 7.33329C9.33317 8.43996 8.43984 9.33329 7.33317 9.33329Z"
        fill="white"
        fill-rule="evenodd"
      />
    </mask>
    <g
      mask="url(#mask1_5004_7786)"
    >
      <rect
        fill="currentColor"
        height="66.6667"
        width="66.6667"
        x="-17.3335"
        y="-17.3334"
      />
      <mask
        height="68"
        id="mask1"
        maskUnits="userSpaceOnUse"
        style="mask-type: luminance;"
        width="68"
        x="-18"
        y="-18"
      >
        <rect
          fill="white"
          height="66.6667"
          width="66.6667"
          x="-17.3335"
          y="-17.3334"
        />
      </mask>
      <g
        mask="url(#mask1)"
      />
    </g>
  </svg>
</div>
`;

exports[`Icons renders Star icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="star-icon"
    fill="none"
    height="24"
    viewBox="0 0 16 16"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M11.3038 13.4049C11.5163 13.2442 11.563 12.9799 11.4334 12.6067L10.3502 9.37774L13.1179 7.39268C13.4392 7.16463 13.5688 6.92104 13.4859 6.66707C13.4029 6.41311 13.1593 6.28872 12.7603 6.2939L9.36544 6.31982L8.33403 3.0753C8.20964 2.69695 8.02306 2.5 7.75355 2.5C7.48922 2.5 7.30263 2.69695 7.17824 3.0753L6.14684 6.31982L2.75202 6.2939C2.35294 6.28872 2.10934 6.41311 2.02641 6.66707C1.9383 6.92104 2.07306 7.16463 2.3944 7.39268L5.16208 9.37774L4.07885 12.6067C3.94928 12.9799 3.99592 13.2442 4.20842 13.4049C4.42092 13.5707 4.69044 13.5137 5.01178 13.2805L7.75355 11.2643L10.5005 13.2805C10.8218 13.5137 11.0862 13.5707 11.3038 13.4049Z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Icons renders Star icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="star-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 16 16"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M11.3038 13.4049C11.5163 13.2442 11.563 12.9799 11.4334 12.6067L10.3502 9.37774L13.1179 7.39268C13.4392 7.16463 13.5688 6.92104 13.4859 6.66707C13.4029 6.41311 13.1593 6.28872 12.7603 6.2939L9.36544 6.31982L8.33403 3.0753C8.20964 2.69695 8.02306 2.5 7.75355 2.5C7.48922 2.5 7.30263 2.69695 7.17824 3.0753L6.14684 6.31982L2.75202 6.2939C2.35294 6.28872 2.10934 6.41311 2.02641 6.66707C1.9383 6.92104 2.07306 7.16463 2.3944 7.39268L5.16208 9.37774L4.07885 12.6067C3.94928 12.9799 3.99592 13.2442 4.20842 13.4049C4.42092 13.5707 4.69044 13.5137 5.01178 13.2805L7.75355 11.2643L10.5005 13.2805C10.8218 13.5137 11.0862 13.5707 11.3038 13.4049Z"
      fill="currentColor"
      fill-rule="evenodd"
    />
  </svg>
</div>
`;

exports[`Icons renders UpArrow icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="up-arrow-icon"
    fill="none"
    height="24"
    viewBox="0 0 25 25"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4201 15.8197L13.9001 9.29967C13.1301 8.52967 11.8701 8.52967 11.1001 9.29967L4.58008 15.8197"
      fill="none"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-miterlimit="10"
      stroke-width="1.5"
    />
  </svg>
</div>
`;

exports[`Icons renders UpArrow icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="up-arrow-icon"
    fill="none"
    height="24"
    viewBox="0 0 25 25"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.4201 15.8197L13.9001 9.29967C13.1301 8.52967 11.8701 8.52967 11.1001 9.29967L4.58008 15.8197"
      fill="currentColor"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-miterlimit="10"
      stroke-width="1.5"
    />
  </svg>
</div>
`;

exports[`Icons renders UserAvatar icon with filled prop false 1`] = `
<div>
  <svg
    data-testid="user-avatar-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M12 22.5C17.7441 22.5 22.5 17.7441 22.5 12C22.5 6.26618 17.7338 1.5 11.9897 1.5C6.25588 1.5 1.5 6.26618 1.5 12C1.5 17.7441 6.26618 22.5 12 22.5ZM18.5265 17.8574C17.4868 16.6221 15.0985 15.5103 12 15.5103C8.90147 15.5103 6.52353 16.6118 5.47353 17.8471C4.09412 16.3029 3.26029 14.2544 3.26029 12C3.26029 7.15147 7.13088 3.25 11.9897 3.25C16.8485 3.25 20.7397 7.15147 20.75 12C20.7603 14.2647 19.9162 16.3132 18.5265 17.8574ZM12 13.7603C13.9765 13.7809 15.5309 12.0926 15.5309 9.87941C15.5309 7.8 13.9765 6.07059 12 6.07059C10.0235 6.07059 8.45882 7.8 8.46912 9.87941C8.47941 12.0926 10.0235 13.7397 12 13.7603Z"
      fill="none"
      fill-rule="evenodd"
      stroke="currentColor"
    />
  </svg>
</div>
`;

exports[`Icons renders UserAvatar icon with filled prop true 1`] = `
<div>
  <svg
    data-testid="user-avatar-icon"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      clip-rule="evenodd"
      d="M12 22.5C17.7441 22.5 22.5 17.7441 22.5 12C22.5 6.26618 17.7338 1.5 11.9897 1.5C6.25588 1.5 1.5 6.26618 1.5 12C1.5 17.7441 6.26618 22.5 12 22.5ZM18.5265 17.8574C17.4868 16.6221 15.0985 15.5103 12 15.5103C8.90147 15.5103 6.52353 16.6118 5.47353 17.8471C4.09412 16.3029 3.26029 14.2544 3.26029 12C3.26029 7.15147 7.13088 3.25 11.9897 3.25C16.8485 3.25 20.7397 7.15147 20.75 12C20.7603 14.2647 19.9162 16.3132 18.5265 17.8574ZM12 13.7603C13.9765 13.7809 15.5309 12.0926 15.5309 9.87941C15.5309 7.8 13.9765 6.07059 12 6.07059C10.0235 6.07059 8.45882 7.8 8.46912 9.87941C8.47941 12.0926 10.0235 13.7397 12 13.7603Z"
      fill="currentColor"
      fill-rule="evenodd"
      stroke="currentColor"
    />
  </svg>
</div>
`;
